packages/react-storage/src/components/StorageBrowser/Views/LocationActionView/UploadControls.tsx

```diff
@@ -315,6 +316,10 @@ export const UploadControls = (): JSX.Element => {
         Start
       </Primary>
       <Destination>{history[history.length - 1].prefix}</Destination>
+      <Overwrite
+        checked={preventOverwrite}
+        handleChange={() => setPreventOverwrite(!preventOverwrite)}
```

Comment:
Prefer providing callbacks to set state handlers for UI events to ensure state is not stale if an end user mashes out on the checkbox, e.g.
```
 setPreventOverwrite(prev => !prev)
```